source/interprocedural_analyses/taint/modelParser.ml

```diff
@@ -23,6 +23,60 @@ open Statement
 open Domains
 open ModelParseResult
 
+module PythonVersion = struct
+  (* Not putting the functions there to prevent circular dependency errors *)
+  include Configuration.PythonVersion
+
+  let parse_from_tuple ~tuple =
+    let parse_element = function
+      | Some { Node.value = Ast.Expression.Expression.Constant const; _ } -> (
+          match const with
+          | Integer integer -> Some (Ok integer)
+          | _ -> Some (Error "Only integer values are allowed inside the version tuple."))
+      | Some _ -> Some (Error "The value in the version tuple must be an integer constant.")
+      | None -> None
+    in
+    (* we need atleast a single value to emulate python tuple comparison *)
+    let major_version =
+      Option.value ~default:(Error "The tuple must not be empty") (parse_element (List.nth tuple 0))
+    in
+    let minor_version = Option.value ~default:(Ok 0) (parse_element (List.nth tuple 1)) in
+    let micro_version = Option.value ~default:(Ok 0) (parse_element (List.nth tuple 2)) in
+    match major_version with
+    | Ok major -> (
+        match minor_version with
+        | Ok minor -> (
+            match micro_version with
+            | Ok micro -> Ok { major; minor; micro }
+            | Error error -> Error error)
+        | Error error -> Error error)
+    | Error error -> Error error
```

Comment:
Looks good! This could be simplified a bit though, using `Core.Result`:
```suggestion
    let open Core.Result in
    Option.value ~default:(Error "The tuple must not be empty") (parse_element (List.nth tuple 0))
    >>= fun major_version ->
    Option.value ~default:(Ok 0) (parse_element (List.nth tuple 1))
    >>= fun minor_version ->
    Option.value ~default:(Ok 0) (parse_element (List.nth tuple 2))
    >>| fun micro_version ->
    { major; minor; micro }
```